pandas/core/nanops.py

```diff
@@ -706,11 +706,20 @@ def nanstd(values, axis=None, skipna=True, ddof=1, mask=None):
     >>> nanops.nanstd(s)
     1.0
     """
+    orig_dtype = values.dtype
+    if is_timedelta64_dtype(values.dtype):
+        # std is well-behaved, but var is not
+        if mask is None:
+            mask = isna(values)
```

Comment:
i think this might be better fixed if we have nanvar call _get_values (which already does this masking & type conversion); its the only routine which doesn't I think (well maybe sem).

Reply:
The problem with that is that we need different behavior for nanstd vs nanvar

Reply:
can you try to address this

Reply:
Well, no.  We have `@disallow("M8", "m8")` on `nanvar`, so if we don't do this here, we'll raise as soon as we call `nanvar` with non-casted values.

Reply:
I am concerned about this code duplicatio of _get_values.

Reply:
I'll double check

Reply:
Yep, turns out we can use _get_values.  updated